packages/cli/src/controller/migrate/manifest/migrate-manifest.controller.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import fs from 'fs';
+import path from 'path';
+import {NETWORK_FAMILY} from '@subql/common';
+import {CommonSubqueryProject} from '@subql/types-core';
+import {upperFirst} from 'lodash';
+import YAML from 'yaml';
+import {findRunnerByNetworkFamily, prepareDirPath, renderTemplate} from '../../../utils';
+import {TemplateKind} from '../../codegen-controller';
+import {graphToSubqlNetworkFamily, networkDsConverters, networkTemplateConverters} from '../constants';
+import {getChainIdByNetworkName} from '../migrate-controller';
+import {MigrateDatasourceKind, SubgraphDataSource, SubgraphProject, SubgraphTemplate, ChainInfo} from '../types';
+
+const PROJECT_TEMPLATE_PATH = path.resolve(__dirname, '../../../template/project.ts.ejs');
+
+export function readSubgraphManifest(inputPath: string): SubgraphProject {
+  if (!fs.existsSync(inputPath)) {
+    throw new Error(`Subgraph manifest ${inputPath} is not exist`);
+  }
+  const subgraphManifest: SubgraphProject = YAML.parse(fs.readFileSync(inputPath, 'utf8'));
+  return subgraphManifest;
+}
+
+export function extractNetworkFromManifest(subgraphProject: SubgraphProject): ChainInfo {
+  const firstDsKind = subgraphProject.dataSources[0].kind;
+  const firstDsNetwork = subgraphProject.dataSources[0].network;
+  if (!firstDsKind || !firstDsNetwork) {
+    throw new Error(`Subgraph dataSource kind or network not been found`);
+  }
```

Comment:
What happens if any ds > 0 is for a different network?

Reply:
I don't think this legal in their manifest, I can add an assertion here too, we don't support this either